eth/handler.go

```diff
@@ -338,6 +335,32 @@ func (pm *ProtocolManager) handle(p *peer) error {
 		p.Log().Info("Ethereum handshake failed", "err", err)
 		return err
 	}
+	forcePeer := false
+	if handler, ok := pm.engine.(consensus.Handler); ok {
+		isValidator, err := handler.Handshake(p)
+		if err != nil {
+			p.Log().Warn("Istanbul handshake failed", "err", err)
+			return err
+		}
+		forcePeer = isValidator
+		p.Log().Trace("Peer completed Istanbul handshake", "forcePeer", forcePeer)
+	}
+	// Ignore max peer and max inbound peer check if:
+	//  - this is a trusted or statically dialed peer
+	//  - the peer is from from the proxy server (e.g. peers connected to this node's internal network interface)
+	//  - forcePeer is true
+	if !forcePeer {
+		isStaticOrTrusted := p.Peer.Info().Network.Trusted || p.Peer.Info().Network.Static
```

Comment:
Just added! Great idea

I still only make the check inside the `if !forcePeer {` block because there isn't a guarantee that the new peer is trusted just yet (because `AddTrustedPeer` involves sending to a channel that's read by the server event loop goroutine), but now we don't expose the inbound count outside of the p2p server